### ISSUE TITLE
Skipping PR merge, required label missing: automerge but the label exists.

### ISSUE THREAD

**Issue by javimox** [pascalgn/automerge-action#95]
Hi,

I am not able to use the automerge action and probably is because I am missing something.. but I can't figure out what. This is the _error_ I get:

`INFO  Skipping PR merge, required label missing: automerge`

The PR has the label `automerge`, though: `@moxbot added the automerge label 2 hours ago`

## Creating PR
This is how the PR is automatically created and the label is added. This part is working fine:
```
    - name: Create Pull Request
      id: cpr
      uses: peter-evans/create-pull-request@v2
      with:
        token: ${{ secrets.PA_TOKEN }}
        branch: test
        labels: automerge
```

## Trying to merge the PR
This is the workflow I have where I use **automerge-action**:
```
on:
  pull_request:
    types: [opened]
jobs:
  automerge:
    runs-on: ubuntu-latest
    steps:
      - name: automerge
        uses: "pascalgn/automerge-action@4536e8847eb62fe2f0ee52c8fa92d17aa97f932f"
        env:
          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
```

This is the output of automerge-action:
```
Run moxbot/automerge-action@4536e8847eb62fe2f0ee52c8fa92d17aa97f932f
/usr/bin/docker run --name c20174cae8ee83db443ea9f34943311fcf44_db0144 --label 87c201 --workdir /github/workspace --rm -e GITHUB_TOKEN -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/qba/qba":"/github/workspace" 87c201:74cae8ee83db443ea9f34943311fcf44
INFO  Event name: pull_request
INFO  Skipping PR update, required label missing: automerge
INFO  Skipping PR merge, required label missing: automerge
```

Any help is welcome.

**Comment by javimox** [comment]
##TL;DR
If I use `types: labeled` instead of `types: opened` in the workflow of the automerge, then it works as expected.
```
on:
  pull_request:
    types: [labeled]
jobs:
  automerge:
    runs-on: ubuntu-latest
    steps:
      - name: automerge
        uses: "pascalgn/automerge-action@4536e8847eb62fe2f0ee52c8fa92d17aa97f932f"
        env:
          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
```

I also tried another GitHub Action for creating PRs and the result was the same: `required label missing: automerge`.
Because it was automerging when I created the PRs myself, I thought it would be a problem of timing. So my guess is that in the moment the PR was opened it didn't have the label `automerge` yet but automerge was triggered. 

Here there is an example:
https://github.com/javimox/test-pr-automerge/pull/9

https://github.com/javimox/test-pr-automerge/blob/master/.github/workflows/automerge.yml

I am going to close the issue, as I gues there's nothing else to do here.

**Comment by pascalgn** [comment]
Maybe just a clarification: yes, this is a shortcoming (IMO) of GH's core API. When a PR is created, you cannot pass labels, see for example https://developer.github.com/v3/pulls/#create-a-pull-request. This means the PR is first created, actions are triggered and _then_ the labels are assigned. You've already figured that out, just wanted to give a reference to the docs 👍
